### PR TITLE
Fix: gray screen flash on nav map

### PR DESCRIPTION
**Pull request by popuz**
## What does this PR change?
Fixes https://github.com/decentraland/unity-renderer/issues/2071

- Add a small controller to disable/enable white background when loading NavMap
- Refactored ExploreV2Controller/View to reduce duplication of the code
- Move responsibility of changing Model from ExploreV2View to ExploreV2Controller

## How to test the changes?
Test 1
1. Open NavMap via shortcut and observe that it loads without the white splash
2. Switch between other tabs (Explore, Backpack, Settings...) and NavMap via UI or Shorcuts and observe no white splash when opening/closing NavMap

Test 2
1. Open NavMap
2. Close ExploreUI via Close button on the right hand side
3. Open Settings via shortcut (P)
4. Observe that Settings are opening on the white background (and you don't see 3d world behind it)

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentController.cs [hunk 1]
```diff
@@ -13,27 +11,33 @@
...
     internal float emotesHUDCloseTime = 0f;
+    internal IExploreV2Analytics exploreV2Analytics;
+    internal MouseCatcher mouseCatcher;
+    internal IPlacesAndEventsSectionComponentController placesAndEventsSectionController;
     internal float playerInfoCardHUDCloseTime = 0f;
-    internal float chatInputHUDCloseTime = 0f;
-    internal List<RealmRowComponentModel> currentAvailableRealms = new List<RealmRowComponentModel>();
+
+    private Dictionary<BaseVariable<bool>, ExploreSection> SectionsByInitVar;
+    private Dictionary<BaseVariable<bool>, ExploreSection> SectionsByVisiblityVar;
```
Comment (sandrade-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentController.cs:28]: Lets follow the same variables naming style as the rest of the code:
```suggestion
    private Dictionary<BaseVariable<bool>, ExploreSection> sectionsByInitVar;
    private Dictionary<BaseVariable<bool>, ExploreSection> sectionsByVisiblityVar;
```

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentController.cs [hunk 2]
```diff
@@ -56,10 +60,23 @@ public class ExploreV2MenuComponentController : IExploreV2MenuComponentControlle
     internal BaseVariable<bool> emotesVisible => DataStore.i.HUDs.emotesVisible;
     internal BaseVariable<bool> chatInputVisible => DataStore.i.HUDs.chatInputVisible;
     internal BooleanVariable playerInfoCardVisible => CommonScriptableObjects.playerInfoCardVisibleState;
-    internal MouseCatcher mouseCatcher;
+
+    private Dictionary<ExploreSection, (BaseVariable<bool> initVar, BaseVariable<bool> visibilityVar)> SectionsVariables =>
```
Comment (sandrade-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentController.cs:64]:
```suggestion
    private Dictionary<ExploreSection, (BaseVariable<bool> initVar, BaseVariable<bool> visibilityVar)> sectionsVariables =>
```

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentController.cs [hunk 3]
```diff
@@ -89,106 +106,56 @@ public void Initialize()
...
+            sectionsVariables.visibilityVar.OnChangeWithSenderInfo += OnSectionVisiblityChanged;
+            OnSectionVisiblityChanged(sectionsVariables.visibilityVar, sectionsVariables.visibilityVar.Get());
+        }
+
+        ConfigureOtherUIDependencies();
 
         isInitialized.Set(true);
-        
+
+        DataStore.i.exploreV2.currentSectionIndex.Set((int)DEFAULT_SECTION, false);
```
Comment (sandrade-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentController.cs:128]:
```suggestion
        currentSectionIndex.Set((int)DEFAULT_SECTION, false);
```

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Common/ShowHideAnimator/ShowHideAnimator.cs [hunk 4]
```diff
@@ -1,21 +1,20 @@
-using System;
+using JetBrains.Annotations;
```
Comment (sandrade-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Common/ShowHideAnimator/ShowHideAnimator.cs:1]: I'm not sure if we should be adding this kind of `JetBrains` dependencies to an open-source project... I think we should discuss it with the team.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Common/ShowHideAnimator/ShowHideAnimator.cs [hunk 5]
```diff
@@ -42,6 +39,17 @@ private int visibleParamHash
         }
     }
 
+    private void OnEnable()
+    {
+        if ( hideOnEnable )
+        {
+            Hide(true);
+        }
+    }
```
Comment (sandrade-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Common/ShowHideAnimator/ShowHideAnimator.cs:48]: I'm seeing the only changes you applied to this `ShowHideAnimator` class have been about to change the order of some things, right?

**Review by sandrade-dcl — APPROVED** [review]
Code reviewed and build tested... I have been testing the whole Explore menu and it seems it's working fine 👍 As we already talked during our last meeting, please make sure that all the metrics (analytics) are being sent in the same way as before.

The code of the `ExploreV2MenuComponentController` class has become much simpler, good work Vitaly! :)

I only left some little corrections, please take a look to them.

And, as final annotation, I noticed that some of your changes were provoked by the automatic formatter of your IDE (Rider I guess). Lets be careful with that because in some cases, when we only need to apply a little change on a PR, this is changing a lot of other parts of the code (for example about methods/variables sorting) and make the PR much bigger than it is, complicating a bit its review.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Common/ShowHideAnimator/ShowHideAnimator.cs (hunk 4 above)
Comment (popuz) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Common/ShowHideAnimator/ShowHideAnimator.cs:1]: as discussed, researched, and answered from JetBrains - it is also OpenSource and under MIT License. [Here ](https://rider-support.jetbrains.com/hc/en-us/community/posts/7665176733458-Is-it-allowed-to-use-JetBrains-Annotations-in-the-OpenSource-project-) is they reply to my question in the forum 
for team discussion with Alex, Kinerius, Nicco - follow this [link to slack thread](https://decentralandteam.slack.com/archives/C03G496L6NN/p1663083874535079)

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentController.cs (hunk 1 above)
Comment (popuz) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentController.cs:28]: good catch on private fields naming notation. I don't think it was my IDE mistake, but more probably I messed up with the naming 😅

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Common/ShowHideAnimator/ShowHideAnimator.cs (hunk 5 above)
Comment (popuz) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Common/ShowHideAnimator/ShowHideAnimator.cs:48]: Mainly, I moved variables which was in-between of public-private methods to be before all methods. Also tried to move events just after properties, and added [UsedImplicitly] attribute. But maybe it was too early and we need to agree on our code organization first...

**Comment by popuz** [comment]
Thank you @sandrade-dcl for the detailed review. When comparing the Analytics message sent (which I catch up via `Debug.Logs `inside `GenericAnalytics.SendAnalytic()`), we can see that the new approach reduces the wrong `Show()` of the previous section and also double call to `Show()` of the new section. Compare call at [12:56:49] from new to the same call at [12:57:26] from old - you can observe that `Map visibility` was `True `for short period (previously chosen section) and `Settings` (newly opened section) was called twice in this frame. 
image

**Comment by popuz** [comment]
however, this issue was only related to the scenario - closing from UI and opening from shortcut a new section. When comparing switching between the sections in UI, old behavior was stable and analytics send well in both - new and old versions of ExploreV2
image

**Comment by sandrade-dcl** [comment]
> Thank you @sandrade-dcl for the detailed review. When comparing the Analytics message sent (which I catch up via `Debug.Logs `inside `GenericAnalytics.SendAnalytic()`), we can see that the new approach reduces the wrong `Show()` of the previous section and also double call to `Show()` of the new section. Compare call at [12:56:49] from new to the same call at [12:57:26] from old - you can observe that `Map visibility` was `True `for short period (previously chosen section) and `Settings` (newly opened section) was called twice in this frame.

Thank you @popuz!! Good work!! 👍

**Review by kwssbocian — APPROVED** [review]
Issue appears to be fixed.
✔️Opening up any of the UI menus (Explore, Backpack, Map and Settings) does not cause the screen to be briefly lit up by a blank white background before loading up the menu.
✔️Settings menu appears with a white background without any 3d world behind it.

**Pull request merged**